py_zipkin/_encoding_helpers.py

```diff
@@ -409,6 +455,48 @@ def encode_span(self, span_builder):
 
         return encoded_span
 
-    def encode_queue(self, queue):
-        """Concatenates the list to a JSON list"""
-        return '[' + ','.join(queue) + ']'
+
+class _V2JSONEncoder(_BaseJSONEncoder):
+    """JSON encoder for V2 spans."""
+
+    def encode_span(self, span_builder):
+        """Encodes a single span to JSON."""
+        span = span_builder.build_v2_span()
+
+        json_span = {
+            'traceId': span.trace_id,
+            'name': span.name,
+            'id': span.id,
+        }
+
+        if span.parent_id:
+            json_span['parentId'] = span.parent_id
+        if span.timestamp:
+            json_span['timestamp'] = int(span.timestamp * 1000000)
+        if span.duration:
+            json_span['duration'] = int(span.duration * 1000000)
+        if span.kind and span.kind.value is not None:
+            json_span['kind'] = span.kind.value
+        if span.local_endpoint:
+            json_span['localEndpoint'] = self._create_json_endpoint(
+                span.local_endpoint,
+            )
+        if span.remote_endpoint:
+            json_span['remoteEndpoint'] = self._create_json_endpoint(
+                span.remote_endpoint,
+            )
+        if span.tags and len(span.tags) > 0:
+            json_span['tags'] = span.tags
+
+        if span.annotations:
+            json_span['annotations'] = [
+                {
+                    'timestamp': int(timestamp * 1000000),
+                    'value': key,
+                }
+                for key, timestamp in span.annotations.items()
+            ]
+
+        encoded_span = json.dumps(json_span)
```

Comment:
does this part serialize debug or shared when they are true?

Reply:
The latest commit added `shared`. We don't support `debug` at all with any encoding for now